airflow/utils/state.py

```diff
@@ -36,9 +36,12 @@ class TerminalTIState(str, Enum):
     """States that a Task Instance can be in that indicate it has reached a terminal state."""
 
     SUCCESS = "success"
-    FAILED = "failed"
+    FAILED = "failed"  # This state indicates that we attempt to retry.
     SKIPPED = "skipped"  # A user can raise a AirflowSkipException from a task & it will be marked as skipped
     REMOVED = "removed"
+    FAIL_WITHOUT_RETRY = (
+        "fail_without_retry"  # This state is useful for when we want to terminate a task, without retrying.
+    )
```

Comment:
```suggestion
    FAIL_WITHOUT_RETRY = "fail_without_retry"
```

in this case I think the name is sufficient

Reply:
Yeah, removed it